### ISSUE TITLE
TUF: Support fallback on trusted Sigstore target for custom roots.

### ISSUE THREAD

**Issue by asraa** [sigstore/cosign#1224]
**Description**

* This came up when I was testing and demo-ing a BYO TUF root. Currently, we require you sign off on all targets that are needed for verification, including Rekor pubkey, Fulcio root CA, CTFE key. If a project wants to only host their own rekor instance, they will still want to fallback on Sigstore's trusted Fulcio instance.

**Comment by haydentherapper** [comment]
Could we also call this support for multiple TUF roots?

I think we'd want a mode where there's never fallback, if someone never wants to trust the hosted infrastructure. If someone wanted to use the fallback, they could initialize cosign with their custom root and Sigstore root. 

On verification, we'd have to update all expired caches, unless there's some mapping between a signature and its root.

**Comment by rgerganov** [comment]
> I think we'd want a mode where there's never fallback, if someone never wants to trust the hosted infrastructure. If someone wanted to use the fallback, they could initialize cosign with their custom root and Sigstore root.
> 

This is possible now. You can create your own TUF root and then initialize cosign like this:
`cosign initialize --mirror http://localhost:8080/repository --root /tmp/tuf/repository/root.json`

**Comment by haydentherapper** [comment]
That would be for hosting a separate root, correct? I was suggesting we add support for multiple roots, and we don't treat the Sigstore root differently than any other root (besides maybe a special flag to initialize the cosign root). For example, supporting something like:

`cosign initialize --root /tmp/tuf/repository/root.json<delimiter>http://localhost:8080/repository --root /other/root.json<delimiter>http://website:8080/repository --root cosign-root-and-mirror

**Comment by asraa** [comment]
> I was suggesting we add support for multiple roots,

Why would you want multiple TUF roots? You want to combine support for verifying against different roots without having to run cosign initialize between verifications?

**Comment by haydentherapper** [comment]
Yep, exactly. Say you want to trust what's signed by the Sigstore root and also some private root too.
